File: leetcode/CountandSay/countAndSay.java
public class Solution {
    public String countAndSay(int n) {
        // Start typing your Java solution below
        // DO NOT write main() function
        String seq = n > 0 ? "1" : "0";
        while (--n > 0) {
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < seq.length(); i++) {
                int j = i;
                while (j < seq.length() && seq.charAt(j) == seq.charAt(i)) j++;
                sb.append(j - i);
                sb.append(seq.charAt(i));
                i = j - 1;
            }
            seq = sb.toString();
        }
        return seq;
    }
}

File: leetcode/CountandSay/countAndSay.cpp
class Solution {
public:
    string helper(string str) {
        string ans = "";
        int cnt = 0, idx = 0, size = str.size();
        while (idx < size) {
            while (idx + 1 < size && str[idx + 1] == str[idx]) {
                cnt++;
                idx++;
            }
            ans += to_string(cnt + 1) + str[idx];
            cnt = 0;
            idx++;
        }
        return ans;
    }
    string countAndSay(int n) {
        string str = "1";
        while (--n > 0) str = helper(str);
        return str;
    }
};
